cirq/circuits/qasm_output.py

```diff
@@ -52,14 +53,24 @@ def _qasm_(self,
                args: protocols.QasmArgs) -> str:
         args.validate_version('2.0')
         return args.format(
-                'u3({0:half_turns},{1:half_turns},{2:half_turns}) {3};\n',
-                self.theta, self.phi, self.lmda, qubits[0])
+            'u3({0:half_turns},{1:half_turns},{2:half_turns}) {3};\n',
+            self.theta, self.phi, self.lmda, qubits[0])
 
     def __repr__(self) -> str:
         return 'cirq.QasmUGate({}, {}, {})'.format(self.lmda,
                                                    self.theta,
                                                    self.phi)
 
+    def _unitary_(self) -> np.ndarray:
+        return np.array([[np.exp(-1j * (self.phi + self.lmda)/2) \
+                          * np.cos(self.theta/2),
```

Comment:
Simplify this by assigning variables for common expressions beforehand.

Alternatively, you may be able to simply take the matrix product of `Rz(a * pi) Rx(b * pi) Rz(c * pi)`.